andela/ah-the-unsullied
# #162949229 Estimate an article's read time

#### What does this PR do?
Estimates an article's read time
#### Description of Task to be completed?
- Add readtime as a dependency in requirements.txt
- Add a read_time field in articles model
- Add a read_time serializer method field in the article serializer class
#### How should this be manually tested?
1. Clone the repo
2. Checkout to the current branch
3. Create a virtual environment and install the requirements
```
- virtualenv <name-of-env>
- source <name-of-env>
- pip install -r requirements.txt
```
4. Create a .env file at the root folder and add the following information:
```
     export DATABASE_URL='postgres://USER:PASSWORD@HOST:PORT/NAME'
     export SECRET_KEY="<your_secret_key_here>"
     export HOST_PASSWORD="<Gmail_Password>"
     export HOST_USER="<Gmail_Username>"
     export SMTP_HOST='smtp.gmail.com'
     export EMAIL_PORT='587'
     export FACEBOOK_KEY='<facebook-key>'
     export FACEBOOK_SECRET='<facebook-secret-key>'
     export TWITTER_KEY='<the-twitter-key>'
     export TWITTER_SECRET_KEY='<twitter-secret-key>'
     export GOOGLE_KEY='<google-key>'
     export GOOGLE_SECRET='<google-secret-key>'
     export ACCESS_TOKEN='<access_token>'
     export SECRET_TOKEN='<secret_token>'
```
5. Run the migrations : `./manage.py migrate`
6. Run the server: `./manage.py runserver`
7. Register a user and create an article
8. Try to get the article. You should be able to see the estimated read time in the response
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#162949229](https://www.pivotaltracker.com/story/show/162949229)
#### Screenshots (if appropriate)
<img width="1440" alt="screenshot 2019-01-30 at 17 27 32" src="https://user-images.githubusercontent.com/23090268/51987817-57f68e00-24b4-11e9-9374-15fa0b4c5aa2.png">
